base/irrationals.jl

```diff
@@ -153,6 +153,8 @@ zero(::Type{<:AbstractIrrational}) = false
 one(::AbstractIrrational) = true
 one(::Type{<:AbstractIrrational}) = true
 
+sign(x::AbstractIrrational) = ifelse(x < zero(x), -1.0, 1.0)
```

Comment:
Why not return `±1` (`Int`) instead of `±1.0`?

For example, you'd ideally like `sign(pi) * 2.3f0` to return a `Float32`, not a `Float64`.

Reply:
Because  -pi returns Float64

Reply:
Feels to me that returning integers is better.

Reply:
if we return an integer here, then there will be three different types among:
- `sign(pi)`
- `zero(pi)`
- `-pi`

Reply:
If integer is better here, wouldn't it then also be better for `sign(::Float64)`? (No, I am not suggesting to make this change, it'd surely be breaking)

Reply:
yeah, sign should be Int8 for everything really